lib/modules/ai_bot/site_settings_extension.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module DiscourseAi::AiBot::SiteSettingsExtension
+  module ClassMethods
+    def ai_bot_enabled=(val)
```

Comment:
Rather than doing these overrides, you can rely on the plugin API `on(:site_setting_changed)` event:

https://github.com/discourse/discourse/blob/a2000a3559eef9ef17d83ee79cf9a3319eb5df24/plugins/chat/plugin.rb#L259-L269

Reply:
It gives you the setting name, old value, and new value.

Reply:
You could still keep a separate class though for `enable_or_disable_ai_bots` since you need to run that from the seed too

Reply:
hmmm maybe, this is cleaner will read through it

Reply:
I think in general it's nicer to dogfood the plugin APIs rather than patch things in this way

Reply:
the trouble though with the API is that it is called for every setting change... I only want to call this on particular changes

Reply:
It's called on every setting change either way, you just need to skip if the setting name doesn't match the one you need. If we're reluctant to use that plugin API it probably indicates it's not a correct design for it, like plugins should be able to register which particular settings they want change events for